packages/wallet/ui/src/components/ProvisionDialog.jsx

```diff
@@ -0,0 +1,123 @@
+import CircularProgress from '@mui/material/CircularProgress';
+import Dialog from '@mui/material/Dialog';
+import DialogTitle from '@mui/material/DialogTitle';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogActions from '@mui/material/DialogActions';
+import Button from '@mui/material/Button';
+import Box from '@mui/material/Box';
+import Link from '@mui/material/Link';
+import { useState } from 'react';
+
+import { withApplicationContext } from '../contexts/Application';
+
+const steps = {
+  INITIAL: 0,
+  AWAITING_APPROVAL: 1,
+  IN_PROGRESS: 2,
+};
+
+const errors = {
+  NO_SIGNER: 'Cannot sign a transaction in read only mode, connect to keplr.',
+};
+
+const ProvisionDialog = ({ onClose, open, address, href, keplrConnection }) => {
+  const [currentStep, setCurrentStep] = useState(steps.INITIAL);
+  const [error, setError] = useState('');
+
+  const provisionWallet = async () => {
+    const {
+      signers: { interactiveSigner },
+    } = keplrConnection;
+    if (!interactiveSigner) {
+      setError(errors.NO_SIGNER);
+      return;
+    }
+
+    setError('');
```

Comment:
why is this necessary? seems an odd place for it. do you want to reset the error when the user initiates a new action?

why use empty string instead of null or undefined?

Reply:
Yea we want to reset the error when the user initiates a new action.

I sometimes get in the habit of using empty string because certain input elements don't play well with null, but no reason otherwise. Typically I never set things to `undefined` explicitly but maybe that's okay as well.

Reply:
> Yea we want to reset the error when the user initiates a new action.

Then please put it where they initiate the action. Either at the top of this function or sooner.

> certain input elements don't play well with null, but no reason otherwise

Please use `null` for "no error". Empty string could be an error that just doesn't have any description. If there are elements where this causes problems I'd like to fix them

Reply:
Changed to use null instead of empty string

I didn't want to have two setErrors in the case of the early return above. I just split this into another function to make it more readable if that's any better?